GridTariffApi.Tests/StartupTasks/PrepareDatabaseStartupTaskTests.cs

```diff
@@ -0,0 +1,115 @@
+            _serviceProvider = services.BuildServiceProvider();
+            _prepareDatabaseStartupTask = new PrepareDatabaseStartupTask(_serviceProvider);
+        }
+
+        [Fact]
+        public async Task ExecuteTest()
+        {
+            const string jsonPayload = "abc123";
+            const string version = "1.0";
+
+            var services = new ServiceCollection();
+            services.AddDbContext<ElviaDbContext>(u => u.UseInMemoryDatabase(databaseName: "ExecuteTest"));
+            var serviceProvider = services.BuildServiceProvider();
+
+            var mockPrepareDatabaseStartupTask = new Mock<PrepareDatabaseStartupTask>(serviceProvider);
+            mockPrepareDatabaseStartupTask.Setup(x => x.GetPricesPayload()).Returns(Task.FromResult(jsonPayload));
+            mockPrepareDatabaseStartupTask.CallBase = true;
+
+            await mockPrepareDatabaseStartupTask.Object.Execute();
+            var elviaDbContext = serviceProvider!.GetRequiredService<ElviaDbContext>();
+
+            var elviaPrices = await elviaDbContext.PriceStructure.Include(x => x.Company).FirstOrDefaultAsync();
+            Assert.Equal(1, await elviaDbContext.PriceStructure.CountAsync());
+            Assert.NotNull(elviaPrices);
+            Assert.Equal(jsonPayload, elviaPrices.JsonPayload);
+            Assert.Equal(version, elviaPrices.JsonVersion);
+            Assert.Equal(DateTimeOffset.UtcNow.Date, elviaPrices.LastUpdatedUtc.Date);
+            Assert.NotNull(elviaPrices.Company);
+            Assert.Equal(1, await elviaDbContext.Company.CountAsync());
+
+            await mockPrepareDatabaseStartupTask.Object.Execute();
+            Assert.Equal(1, await elviaDbContext.PriceStructure.CountAsync());
+            Assert.Equal(1, await elviaDbContext.Company.CountAsync());
+        }
+
+
+        [Fact]
+        public async Task GetElviaPricesHitTest()
+        {
+            Setup();
+
+            var elviaDbContext = _serviceProvider!.GetRequiredService<ElviaDbContext>();
+            Assert.Null(await elviaDbContext.Company.FirstOrDefaultAsync());
+            Assert.Null(await elviaDbContext.PriceStructure.FirstOrDefaultAsync());
+
+            var elviaCompany = await _prepareDatabaseStartupTask!.GetElviaCompany(elviaDbContext);
+            var priceStructure = await _prepareDatabaseStartupTask.GetElviaPrices(elviaDbContext, elviaCompany);
+            Assert.Equal(elviaCompany.OrgNumber, priceStructure.Company.OrgNumber);
+            Assert.Equal(1, await elviaDbContext.PriceStructure.CountAsync());
+            await _prepareDatabaseStartupTask.GetElviaPrices(elviaDbContext, elviaCompany);
+            Assert.Equal(elviaCompany.OrgNumber, priceStructure.Company.OrgNumber);
+            Assert.Equal(1, await elviaDbContext.PriceStructure.CountAsync());
+        }
+
+
+        [Fact]
+        public async Task GetElviaPricesMissTest()
```

Comment:
Denne er kanskje ikke ferdig? Tester det samme som GetElviaPricesHitTest, bare litt mindre.

Reply:
Mora eller dattera?
MissTest har ansvar for å teste hva som skjer når prisstruktur ikke er tidligere persistert.
HitTest har ansvar for å teste hva som skjer når prisstruktur allerede er persistert.

Reply:
Har slått sammen disse til felles test, dvs fjernet GetElviaPricesMissTest